v2/test/presubmit.sh

```diff
@@ -26,12 +26,19 @@ GCR_ROOT="gcr.io/${PROJECT}/${COMMIT_SHA}/v2-sample-test"
 # This is kfp-ci endpoint.
 HOST=${HOST:-"https://$(curl https://raw.githubusercontent.com/kubeflow/testing/master/test-infra/kfp/endpoint)"}
 
+if [[ -z "${PULL_NUMBER}" ]]; then
+  KFP_PACKAGE_PATH='git+https://github.com/kubeflow/pipelines\#egg=kfp&subdirectory=sdk/python'
+else
+  KFP_PACKAGE_PATH='git+https://github.com/kubeflow/pipelines@refs/pull/${PULL_NUMBER}/merge\#egg=kfp&subdirectory=sdk/python'
```

Comment:
just realized another caveat, current KFP test infra tests by:
1. merge the PR to master branch
2. continue the test using merged code

this guarantees every PR is tested against latest status.

However, this will use the PR itself's code for the container runtime package.

This might not be a big deal, but just keep that in mind.

Reply:
Nevermind, I just realized the pull/merge url is already merged version of the pull request.
so this concern does not exist
https://stackoverflow.com/a/50095199/8745218